File: week3/lib/prime.c
#include <stdlib.h>

#include "prime.h"

void calculate_primes(int *data, int kmax)
{
    int i,j,k,n;
    int *p;

    p = calloc(kmax, sizeof(int));

    k = 0; n = 2; j = 0;
    while (k < kmax) {
        i = 0;
        while ((i < k) && (n % p[i] != 0))
            i = i + 1;
        if (i == k) {
            p[k] = n;
            k = k + 1;
            data[j++] = n;
        }
        n += 1;
    }

    free(p);
}

File: week3/lib/prime.py
def primes(kmax):
    if kmax > 100000:
        kmax = 100000
    k = 0
    n = 2
    p = [0] * kmax
    while k < kmax:
        i = 0
        while i < k and n % p[i] != 0:
            i = i + 1
        if i == k:
            p[k] = n
            k = k + 1
        n = n + 1
    return p
